### PR TITLE
Added example subsection to _cmake_reference

### PR DESCRIPTION
**Pull request by mristin**
This update is related to the issue #716 .

### CODE REVIEW

File: reference/build_helpers/cmake.rst [hunk 1]
```diff
@@ -292,3 +292,42 @@ Environment variables
...
 :ref:`CMAKE RELATED VARIABLES<cmake_related_variables>` section.
+
+Example
+-------
+The following example of ``conanfile.py`` shows you how to manage a project with conan and CMake.
+
+.. code-block:: python
+
+    import conans
+    import conans.tools
```
Comment (danimtb) [inline reference/build_helpers/cmake.rst]: `from conans import ConanFile, CMake, tools`

File: reference/build_helpers/cmake.rst [hunk 2]
```diff
@@ -292,3 +292,42 @@ Environment variables
...
+
+    class SomePackage(conans.ConanFile):
+        name = "SomePackage"
+        version = "1.0.0"
+        settings = "os", "compiler", "build_type", "arch"
+        options = {"shared": [True, False]}
```
Comment (danimtb) [inline reference/build_helpers/cmake.rst]: Remove not relevant information for this example: options & default_options, keep_imports, requires

File: reference/build_helpers/cmake.rst [hunk 3]
```diff
@@ -292,3 +292,38 @@ Environment variables
...
+        # build(), test() and package()
+        cmake.definitions["SOME_DEFINITION_NAME"] = "On"
+
+        return cmake
+
+    def build(self):
+        cmake = self.configure_cmake()
+        cmake.build()
+
+    def test(self):
```
Comment (memsharded) [inline reference/build_helpers/cmake.rst]: This is good for ``test_package`` functionality, but this code is not run for normal packaging.
Maybe many users want to make sure to run the tests in the ``build()`` method, so they are run for every build of the package, and abort the packaging if they fail. In that case, the ``test()`` method will be removed, and the build would be:

```python
   def build(self):
        cmake = self.configure_cmake()
        cmake.build()
        cmake.test()
```

File: reference/build_helpers/cmake.rst [hunk 4]
```diff
@@ -292,3 +292,38 @@ Environment variables
...
+        cmake = self.configure_cmake()
+        self.run("make check")
```
Comment (memsharded) [inline reference/build_helpers/cmake.rst]: The ``make check`` is too Makefile specific. I think the general approach should be ``cmake.test()``. If for some reason it is not working, please report.

File: reference/build_helpers/cmake.rst (hunk 3 above)
Comment (marko-ristin-parquery) [reply reference/build_helpers/cmake.rst]: I added the example commands for two scenarios: one that you suggest (run unit tests after the build), and another command to run a custom make command.

File: reference/build_helpers/cmake.rst (hunk 4 above)
Comment (marko-ristin-parquery) [reply reference/build_helpers/cmake.rst]: See the comment re this change above.

**Comment by mristin** [comment]
(Sorry, I closed the issue unintentionally.)

**Comment by mristin** [comment]
Hi @memsharded,
Just a kind reminder in case the last commit got lost in the backlog -- would you mind reviewing?

**Comment by mristin** [comment]
Hi @memsharded ,
I'm staring at the code but can't see where I missed the `cmake.configure()` :-(

If you compare against the master branch:

```python
    def configure_cmake(self):
        ...

    def build(self):
        cmake = self.configure_cmake()
        ...

    def package(self):
        cmake = self.configure_cmake()
        ...
```

Could you please point me?

File: reference/build_helpers/cmake.rst [hunk 5]
```diff
@@ -292,3 +292,40 @@ Environment variables
...
+        settings = "os", "compiler", "build_type", "arch"
+        generators = "cmake"
+
+    def configure_cmake(self):
+        cmake = CMake(self)
+
+        # put definitions here so that they are re-used in cmake between
+        # build() and package()
+        cmake.definitions["SOME_DEFINITION_NAME"] = "On"
+
```
Comment (memsharded) [inline reference/build_helpers/cmake.rst]: Maybe here, insert the call to ``cmake.configure()``?

Comment (mristin) [reply reference/build_helpers/cmake.rst]: Exactly :smile: added, sorry for the blind spot.

**Comment by memsharded** [comment]
Many thanks!

File: reference/build_helpers/cmake.rst [hunk 6]
```diff
@@ -292,3 +292,41 @@ Environment variables
...
+    from conans import ConanFile, CMake
+
+    class SomePackage(ConanFile):
+        name = "SomePackage"
+        version = "1.0.0"
+        settings = "os", "compiler", "build_type", "arch"
+        generators = "cmake"
+
+    def configure_cmake(self):
+        cmake = CMake(self)
```
Comment (bryceschober) [inline reference/build_helpers/cmake.rst]: IMO, `set_cmake_flags=True` should be mentioned more prominently in the docs, and this is a good opportunity. It's the kind of option that I thought should be default behavior as newbie.

Comment (danimtb) [reply reference/build_helpers/cmake.rst]: It is well documented here https://docs.conan.io/en/latest/reference/build_helpers/cmake.html#constructor and has not raised any particular issue so far on its default value

Comment (bryceschober) [reply reference/build_helpers/cmake.rst]: Sorry, I was off-base on the current documentation context.

File: reference/build_helpers/cmake.rst [hunk 7]
```diff
@@ -292,3 +292,41 @@ Environment variables
...
+
+    from conans import ConanFile, CMake
+
+    class SomePackage(ConanFile):
+        name = "SomePackage"
+        version = "1.0.0"
+        settings = "os", "compiler", "build_type", "arch"
+        generators = "cmake"
+
+    def configure_cmake(self):
```
Comment (johnmcfarlane) [inline reference/build_helpers/cmake.rst:310]: Indentation doesn't look right in this example.

Comment (memsharded) [reply reference/build_helpers/cmake.rst:310]: Definitely wrong, thanks

Comment (memsharded) [reply reference/build_helpers/cmake.rst:310]: https://github.com/conan-io/docs/pull/1954

**Pull request merged**